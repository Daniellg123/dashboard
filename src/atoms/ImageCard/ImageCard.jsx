import { CardMedia } from '@mui/material';

import './ImageCard.css';

function ImageCard({ imgDir, size }) {
    return (
        <CardMedia
            className='img'
            component='img'
            sx={{ width: size }}
            image={imgDir}
            alt={imgDir}
        />
    );
};

export default ImageCard;