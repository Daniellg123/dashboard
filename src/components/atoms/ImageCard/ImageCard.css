.img {
    border-radius: 102.4px;
}