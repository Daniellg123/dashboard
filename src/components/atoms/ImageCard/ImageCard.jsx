import PropTypes from 'prop-types';
import { CardMedia } from '@mui/material';

/**
 * Componente para mostrar una imagen dentro de una card
 */

import './ImageCard.css';

function ImageCard({ imgDir, size }) {
    return (
        <CardMedia
            className='img'
            component='img'
            sx={{ width: size }}
            image={imgDir}
            alt={imgDir}
        />
    );
};

ImageCard.propTypes = {
    /**
     * Ruta de la imagen
     */
    imgDir: PropTypes.string,
    /**
     * Tama√±o de la imagen
     */
    size: PropTypes.number
};

export default ImageCard;