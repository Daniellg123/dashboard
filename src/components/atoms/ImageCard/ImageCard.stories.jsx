import ImageCard from "./ImageCard";

const routeImg = './img/OrganizationChart.png';

export default {
    title: 'Atoms/ImageCard',
    component: ImageCard,
};

const Template = (args) => <ImageCard {...args}/>

export const Primary = Template.bind({});
Primary.args = { imgDir: routeImg, size: 48};