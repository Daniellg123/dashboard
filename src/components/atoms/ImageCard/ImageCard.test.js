import renderer from 'react-test-renderer'
import ImageCard from './ImageCard';

const props = {
    imgDir: './img/logo1.png',
    size: 18
};

it('renders correctly', () => {
    const render = renderer.create(<ImageCard {...props}/>).toJSON();
    expect(render).toMatchSnapshot();
});