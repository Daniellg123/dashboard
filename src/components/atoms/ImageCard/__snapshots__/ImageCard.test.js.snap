// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="./img/logo1.png"
  className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-iqx0p4-MuiCardMedia-root"
  src="./img/logo1.png"
/>
`;
